### PR TITLE
ESQL: Docs: COUNT: add an explanation to the use of the 3VL

### LINKED ISSUE #99954: ESQL: improve COUNT documentation
### Description

At the moment, the COUNT documentation is very slim and the example provided there is the most basic one.
Would be better with a more complex example with multiple `by` keys and, also, to be specific about how `count(column)` deals with `null` keys:

`from employees | stats c = count(gender) by gender`

```
       c       |    gender     
---------------+---------------
0              |null           
57             |M              
33             |F              

```

### PR DESCRIPTION
**Pull request by bpintea**
Add an explanation of why `... OR NULL` is needed with `COUNT()`.

Fixes: #99954

### CODE REVIEW

**Comment by bpintea** [comment]
@elasticmachine update branch

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/Count.java [hunk 1]
```diff
@@ -58,7 +58,9 @@ public class Count extends AggregateFunction implements ToAggregator, SurrogateE
             ),
             @Example(
                 description = "To count the same stream of data based on two different expressions "
```
Comment (ivancea) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/Count.java:60]: I don't understand the `based on two different expressions`. Which 2 different expressions are in that example?
Everything else you added helps a lot btw

**Review by ivancea — APPROVED** [review]

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/Count.java (hunk 1 above)
Comment (bpintea) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/Count.java:60]: `ROW n=1
| STATS COUNT(n > 0 OR NULL), COUNT(n < 0 OR NULL)`
is the example added here (so gt, lt).

Comment (ivancea) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/Count.java:60]: Ah, I thought it was expressions in the same function. I need some extra coffee

**Review by nik9000 — APPROVED** [review]

**Pull request merged**